test/client/records-queue.spec.js

```diff
@@ -0,0 +1,526 @@
+                enketoId,
+                instanceId: 'recordD',
+            } );
+
+            store.survey.set( survey )
+                .then( () => encryptor.encryptRecord( form, recordC ) )
+                .then( encryptedRecordC => {
+                    return records.save( 'set', encryptedRecordC );
+                } )
+                .then( () => {
+                    return formCache.getLastSavedRecord( enketoId );
+                } )
+                .then( () => encryptor.encryptRecord( form, recordD ) )
+                .then( encryptedRecordD => {
+                    return records.save( 'set', encryptedRecordD );
+                } )
+                .then( () => {
+                    return formCache.getLastSavedRecord( enketoId );
+                } )
+                .then( record => {
+                    expect( record ).to.equal( undefined );
+                } )
+                .then( done, done );
+        } );
+    } );
+
+    describe( 'Retrieving records', () => {
+        it( 'gets the record list, excludes the auto-saved records', done => {
+            const autoSavedKey = records.getAutoSavedKey();
+
+            const expectedRecordData = [
+                Object.assign( {}, recordA ),
+                Object.assign( {}, recordB ),
+            ];
+
+            records.save( 'set', recordA )
+                .then( () => {
+                    return records.save( 'set', recordB );
+                } )
+                .then( () => {
+                    return records.getDisplayableRecordList( enketoId );
+                } )
+                .then( records => {
+                    expect( records.length ).to.equal( expectedRecordData.length );
+
+                    records.forEach( record => {
+                        expect( record.instanceId ).not.to.equal( autoSavedKey );
+                    } );
+
+                    expectedRecordData.forEach( ( recordData, index ) => {
+                        const record = records[index];
+
+                        Object.entries( recordData ).forEach( ( [ key, value ] ) => {
+                            expect( record[key] ).to.deep.equal( value );
+                        } );
+                    } );
+                } )
+                .then( done, done );
+        } );
+
+        // This is primarily testing that an empty array is returned as the db.js
+        // types indicate, rather than needing to keep using falsy checks.
+        it( 'gets an empty record list', done => {
+            records.getDisplayableRecordList( enketoId )
+                .then( records => {
+                    expect( Array.isArray( records ) ).to.equal( true );
+                    expect( records.length ).to.equal( 0 );
+                } )
+                .then( done, done );
+        } );
+    } );
+
+    describe( 'Uploading records', () => {
+        it( 'uploads queued records, excluding auto-saved/last-saved and draft records', done => {
```

Comment:
I'd originally kept this in as it was part of the criteria in #137 and felt like a good regression test in case we decide to change the approach, but I agree it feels like irrelevant noise now.